app/src/main/java/org/p2p/wallet/history/repository/local/TransactionDetailsDatabaseRepository.kt

```diff
@@ -1,18 +1,37 @@
 package org.p2p.wallet.history.repository.local
 
 import org.p2p.solanaj.kits.transaction.TransactionDetails
+import org.p2p.wallet.history.model.HistoryTransaction
 import org.p2p.wallet.history.repository.local.mapper.TransactionDetailsEntityMapper
 import org.p2p.wallet.history.repository.local.db.dao.TransactionsDaoDelegate
 
 class TransactionDetailsDatabaseRepository(
     private val daoDelegate: TransactionsDaoDelegate,
     private val mapper: TransactionDetailsEntityMapper,
 ) : TransactionDetailsLocalRepository {
+    private val pendingTransactions = HashMap<String, HistoryTransaction?>()
+
     override suspend fun getTransactions(signatures: List<String>): List<TransactionDetails> {
         return daoDelegate.getTransactions(signatures)
             .let { mapper.fromEntityToDomain(it) }
     }
 
+    override suspend fun savePendingTransaction(txSignature: String, transaction: HistoryTransaction) {
```

Comment:
It's not database repository anymore :( 
InMemory - yes

Reply:
okay, but there is dao delegate )